### PR TITLE
Add GitHub Action tests

### PR DESCRIPTION
**Pull request by ocaisa**

### CODE REVIEW

File: .check_with_codespell.sh [hunk 1]
```diff
@@ -0,0 +1 @@
+codespell --skip='.git,version-specific,scripts' --ignore-words-list=atleast,ninjs,simpy,proovread --quiet-level=2; retVal=$?; if [ $retVal -eq 1 ]; then echo 'Got one expected warning, so success!';  else $(exit $retVal); fi;
```
Comment (ocaisa) [inline .check_with_codespell.sh]: I  couldn't find a way to execute this line in a github action so I had to hide it in a script

File: .github/workflows/doc_build.yml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+      run: |
+        pip install flake8
+        # stop the build if there are Python syntax errors or undefined names
+        flake8 . --count --select=E9,F63,F7,F82 --show-source --statistics
+        # exit-zero treats all errors as warnings. The GitHub editor is 127 chars wide
+        flake8 . --count --exit-zero --max-complexity=10 --max-line-length=127 --statistics
+    - name: Test with codespell
+      run: |
+        pip install codespell
+        bash .check_with_codespell.sh
```
Comment (boegel) [inline .github/workflows/doc_build.yml]: @ocaisa Would be better to move this script into `.github/workflows/`?

File: .github/workflows/doc_build.yml [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    - name: Test with sphinx
+      run: |
+        pip install sphinx
+        READTHEDOCS=1 sphinx-build docs build
+    - name: Test installation
+      run: |
+        pip install .
```
Comment (boegel) [inline .github/workflows/doc_build.yml]: add back comment `# test installation too (using options that don't require to have a modules tool installed)`

File: .github/workflows/doc_build.yml [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Python 3.8
+      uses: actions/setup-python@v1
+      with:
+        python-version: 3.8
+    - name: Lint with flake8
+      run: |
+        pip install flake8
```
Comment (boegel) [inline .github/workflows/doc_build.yml]: add missing comment that we have in `.travis.yml`?

File: .github/workflows/doc_build.yml [hunk 5]
```diff
@@ -0,0 +1,35 @@
+name: Docs build
```
Comment (boegel) [inline .github/workflows/doc_build.yml]: we check more than just docs, we also test the most recent `easybuild` installation end-to-end

**Comment by boegel** [comment]
Thanks @ocaisa!

**Pull request merged**